File: Model/MatchGenerators/Matchday.vb
Public Class Matchday
    Inherits List(Of MatchIndexPair)

    Public Function ContainsPlayerIn(match As MatchIndexPair) As Boolean
        Return Me.Any(Function(m) _
            m.Item1 = match.Item1 OrElse
            m.Item1 = match.Item2 OrElse
            m.Item2 = match.Item1 OrElse
            m.Item2 = match.Item2
        )
    End Function

    Public Function MatchesWithPlayersFrom(match As MatchIndexPair) As IEnumerable(Of MatchIndexPair)
        Return Me.Where(Function(m) _
            m.Item1 = match.Item1 OrElse
            m.Item1 = match.Item2 OrElse
            m.Item2 = match.Item1 OrElse
            m.Item2 = match.Item2
        )
    End Function
End Class

File: POFF.Kicker/MatchGenerators/Matchday.cs
using System.Collections.Generic;
using System.Data;
using System.Linq;

namespace POFF.Kicker.Model;

public class Matchday : List<MatchIndexPair>
{

    public bool ContainsPlayerIn(MatchIndexPair match)
    {
        return this.Any(m => m.Item1 == match.Item1 || m.Item1 == match.Item2 || m.Item2 == match.Item1 || m.Item2 == match.Item2);
    }

    public IEnumerable<MatchIndexPair> MatchesWithPlayersFrom(MatchIndexPair match)
    {
        return this.Where(m => m.Item1 == match.Item1 || m.Item1 == match.Item2 || m.Item2 == match.Item1 || m.Item2 == match.Item2);
    }
}